src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/UnitTests/Controllers/HomeControllerTests.cs

```diff
@@ -1,10 +1,87 @@
 using System;
 using System.Collections.Generic;
+using System.Security.Claims;
+using System.Security.Principal;
 using System.Text;
+using DevAdventCalendarCompetition.Controllers;
+using DevAdventCalendarCompetition.Models.Test;
+using DevAdventCalendarCompetition.Repository.Models;
+using DevAdventCalendarCompetition.Services.Interfaces;
+using DevAdventCalendarCompetition.Services.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+using Xunit;
+using static DevAdventCalendarCompetition.Tests.TestHelper;
 
 namespace DevAdventCalendarCompetition.Tests.UnitTests.Controllers
 {
     public class HomeControllerTests
     {
+        private readonly Mock<IHomeService> _homeServiceMock;
+
+        public HomeControllerTests()
+        {
+            this._homeServiceMock = new Mock<IHomeService>();
+        }
+
+        [Fact]
+        public void Index_CurrentTestDtoIsNull_ReturnsEmptyViewResult()
+        {
+            // Arrange
+            this._homeServiceMock.Setup(x => x.GetCurrentTests()).Returns((List<TestDto>)null);
+            using var controller = new HomeController(this._homeServiceMock.Object);
+
+            // Act
+            var result = controller.Index();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.Model);
+        }
+
+        [Fact]
+        public void Index_UserIsNull_ReturnsAViewResutWithListOfTestDto()
+        {
```

Comment:
@Mejtika I tried to find out solution in internet but unfortunately did not find suitable approach. How would you mock user from httpContext in homeConroller. I am working on test ( Index_UserIsNull_ReturnsAViewResutWithListOfTestDto)

Reply:
The simplest solution would be to create UserService, that we'll then mock in tests. Because we're not using it in the entire controller, we can use method injection rather than constructor Try to do it yourself.
![image](https://user-images.githubusercontent.com/28907927/91188121-39d52c00-e6f1-11ea-9ee7-61f6d8e6c854.png)